src/expr/src/scalar/func.rs

```diff
@@ -1932,18 +1932,22 @@ pub fn date_bin<'a, T>(stride: Interval, source: T, origin: T) -> Result<Datum<'
 where
     T: TimestampLike,
 {
-    let stride_ns = if stride.months != 0 {
-        Err(EvalError::DateBinOutOfRange(
+    if stride.months != 0 {
+        return Err(EvalError::DateBinOutOfRange(
             "timestamps cannot be binned into intervals containing months or years".to_string(),
-        ))
-    } else if stride.duration <= 0 {
-        Err(EvalError::DateBinOutOfRange(
+        ));
+    }
+
+    let stride_ns = match stride.duration_as_chrono().num_nanoseconds() {
```

Comment:
I'm kind of interested in removing chrono from this. Can we have a num_nanoseconds method directly on interval that doesn't go through chrono? Can you do the math to see if the max/min intervals can always be converted safely to i64 nanos? If not, then it should return a Result/Option. Similarly to the above, the interval type should be in the business of safely converting itself to various forms so that callers don't have to remember to do it safely.